From 337e12bf2dbdc3fb5fccbc3072ab97de896d0c6f Mon Sep 17 00:00:00 2001
From: psyafter <psyafter@gmail.com>
Date: Mon, 14 Nov 2016 15:13:42 +0200
Subject: [PATCH] telephony: allow broadcasts from Samsung RIL

RILD triggers message broadcasts using the UNSOL_AM command. These
commands are executed by RILJ. As of Android NG message broadcasts are
only allowed, if the intents are in the protected-broadcast whitelist.
Otherwise the VM will be terminated.

Add the intents used by Samsungs RIL to the protected-broadcast
whitelist.

Change-Id: Idd1bc1c9c68d29ac4a36a9b09a3a5fc5c28af3ee
---
 AndroidManifest.xml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index dbc158c..fbe1993 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -71,6 +71,10 @@
     <protected-broadcast android:name= "com.android.intent.action.IMS_FEATURE_CHANGED" />
     <protected-broadcast android:name= "com.android.intent.action.IMS_CONFIG_CHANGED" />
 
+    <protected-broadcast android:name= "com.samsung.intent.action.WB_AMR" />
+    <protected-broadcast android:name= "android.intent.action.pcmclkctrl" />
+    <protected-broadcast android:name= "android.intent.action.PHONE_EXSTATE_CHANGED" />
+
     <uses-permission android:name="android.permission.BROADCAST_STICKY" />
     <uses-permission android:name="android.permission.CALL_PHONE" />
     <uses-permission android:name="android.permission.CALL_PRIVILEGED" />
-- 
2.7.4

